jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java

```diff
@@ -648,12 +649,16 @@ private ByteBuffer acquireBuffer()
         return _aggregate;
     }
 
-    private void releaseBuffer()
+    private void releaseBuffer(Throwable failure)
     {
         if (_aggregate != null)
         {
+            boolean asyncCancelPossible = _channel.getRequest().getMetaData().getHttpVersion().getVersion() >= HttpVersion.HTTP_2.getVersion();
```

Comment:
minor niggle.... probably resolved by JIT.... we only need to follow those 4 de-references to check the version if there is a failure.

Reply:
Ok, I've inlined the HTTP version check to the right side of the `||`.